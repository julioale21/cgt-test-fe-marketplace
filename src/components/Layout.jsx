import { Outlet } from 'react-router-dom';
import { Header } from '../components/shared/Header';
import { Box } from '@mui/material';

export function Layout() {
  const cartItemsCount = 3;

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
      <Header cartItemsCount={cartItemsCount} />

      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <Outlet />
      </Box>

      <Box component="footer" sx={{ p: 2, textAlign: 'center', mt: 'auto' }}>
        <p>Â© 2025</p>
      </Box>
    </Box>
  );
}
