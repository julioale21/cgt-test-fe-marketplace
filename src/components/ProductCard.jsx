import React from 'react';

import { Link } from 'react-router-dom';

import { Button, Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const ProductCard = ({ product }) => {
  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={product.image}
          alt={product.name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {product.name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {product.description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${product.price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {product.format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions sx={{ justifyContent: 'space-between' }}>
          <Button component={Link} to={`/products/${product.id}`} size="small" color="primary">
            View Details
          </Button>
          <Button size="small" variant="contained" color="primary" startIcon={<ShoppingCartIcon />}>
            Add to Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export { ProductCard };
