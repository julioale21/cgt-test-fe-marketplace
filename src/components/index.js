export { CartHeader } from "./cart/CartHeader";
export { CartItem } from "./cart/CartItem";
export { CartItemList } from "./cart/CartItemList";
export { CartSummary } from "./cart/CartSummary";
export { CartSummaryRow } from "./cart/CartSummaryRow";
export { ContainedButton } from "./buttons/ContainedButton";
export { CustomAlert } from "./shared/CustomAlert";
export { EmptyCart } from "./cart/EmptyCart";
export { Header } from "./shared/Header";
export { Layout } from "./Layout";
export { OrderDetailsSection } from "./checkout/success_modal/OrderDetailsSection";
export { OrderSummary } from "./checkout/OrderSummary";
export { OutlinedButton } from "./buttons/OutlinedButton";
export { PaymentForm } from "./checkout/PaymentForm"
export { PaymentMethodIcon } from "./cart/PaymentMethodIcon";
export { ProductCard } from "./product/ProductCard";
export { ProductDetails } from "./product/ProductDetail";
export { ProductImage } from "./product/ProductImage";
export { ProductList } from "./product/ProductList";
export { ProductSpecs } from "./product/ProductSpecs";
export { ShippingSection } from "./checkout/success_modal/ShippingSection";
export { SuccessHeader } from "./checkout/success_modal/SuccessHeader";
export {PaymentSuccessModal} from "./checkout/success_modal/PaymentSuccessModal";
export {ProductHeader} from "./product/ProductHeader";
