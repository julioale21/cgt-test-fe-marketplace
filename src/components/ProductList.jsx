import React from 'react';
import { Grid } from '@mui/material';
import { ProductCard } from './ProductCard';

const ProductList = ({ products }) => {
  return (
    <Grid container spacing={4}>
      {products.map((product) => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { ProductList };
