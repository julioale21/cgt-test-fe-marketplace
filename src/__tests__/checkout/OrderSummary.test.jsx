import React from 'react';
import { render, screen } from '@testing-library/react';
import { OrderSummary } from '../../components';


const mockItems = [
  {
    id: 1,
    name: 'Test Product 1',
    price: 29.99,
    quantity: 2,
    image: 'test-image-1.jpg'
  },
  {
    id: 2,
    name: 'Test Product 2',
    price: 39.99,
    quantity: 1,
    image: 'test-image-2.jpg'
  },
  {
    id: 3,
    name: 'Test Product 3',
    price: 49.99,
    quantity: 1,
    image: 'test-image-3.jpg'
  }
];

const defaultProps = {
  items: mockItems,
  subtotal: 149.96,
  shipping: 0,
  taxes: 12.5,
  finalTotal: 162.46
};

describe('OrderSummary', () => {
  it('should match snapshot', () => {
    const { container } = render(<OrderSummary {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should display order summary title', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('Order Summary')).toBeInTheDocument();
  });

  it('should display only first two items from the list', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('Test Product 1')).toBeInTheDocument();
    expect(screen.getByText('Test Product 2')).toBeInTheDocument();
    expect(screen.queryByText('Test Product 3')).not.toBeInTheDocument();
  });

  it('should show remaining items count when more than 2 items exist', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('And 1 more items...')).toBeInTheDocument();
  });

  it('should not show remaining items count when 2 or fewer items exist', () => {
    render(<OrderSummary {...{ ...defaultProps, items: mockItems.slice(0, 2) }} />);

    expect(screen.queryByText(/more items/)).not.toBeInTheDocument();
  });

  it('should display correct item quantities', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('Qty: 2')).toBeInTheDocument();
    expect(screen.getByText('Qty: 1')).toBeInTheDocument();
  });

  it('should display correct price calculations', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('$59.98')).toBeInTheDocument();
    expect(screen.getByText('$39.99')).toBeInTheDocument();
  });

  it('should display correct order totals', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('$149.96')).toBeInTheDocument();
    expect(screen.getByText('$0.00')).toBeInTheDocument();
    expect(screen.getByText('$12.50')).toBeInTheDocument();
    expect(screen.getByText('$162.46')).toBeInTheDocument();
  });

  it('should display shipping information', () => {
    render(<OrderSummary {...defaultProps} />);

    expect(screen.getByText('Free Shipping')).toBeInTheDocument();
    expect(screen.getByText('Estimated delivery: 3-5 business days')).toBeInTheDocument();
  });
});
