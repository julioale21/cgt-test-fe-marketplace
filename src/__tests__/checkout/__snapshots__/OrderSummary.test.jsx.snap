// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1rr4qq7"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-2b7ht-MuiPaper-root"
      style="--Paper-shadow: none;"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-19hh0f1-MuiTypography-root"
      >
        Order Summary
      </h5>
      <div
        class="MuiBox-root css-i3pbo"
      >
        <div
          class="MuiBox-root css-6msto9"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded css-15k0rve-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
              src="test-image-1.jpg"
            />
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-13k9d3g-MuiTypography-root"
            >
              Test Product 1
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
            >
              Qty: 
              2
            </p>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1ypmsg6-MuiTypography-root"
          >
            $
            59.98
          </h6>
        </div>
        <div
          class="MuiBox-root css-6msto9"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded css-15k0rve-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
              src="test-image-2.jpg"
            />
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-13k9d3g-MuiTypography-root"
            >
              Test Product 2
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
            >
              Qty: 
              1
            </p>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1ypmsg6-MuiTypography-root"
          >
            $
            39.99
          </h6>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-tmy7vb-MuiTypography-root"
        >
          And 
          1
           more items...
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1ouqunm-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-i3pbo"
      >
        <div
          class="MuiBox-root css-3mllz6"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
          >
            Subtotal
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            $
            149.96
          </p>
        </div>
        <div
          class="MuiBox-root css-3mllz6"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
          >
            Shipping
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            $
            0.00
          </p>
        </div>
        <div
          class="MuiBox-root css-3mllz6"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
          >
            Taxes
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            $
            12.50
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1qu4ks8"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          Total
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          $
          162.46
        </h6>
      </div>
      <div
        class="MuiBox-root css-h5fkc8"
      >
        <div
          class="MuiBox-root css-127k144"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1wtp6o0-MuiSvgIcon-root"
            data-testid="LocalShippingIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 8h-3V4H3c-1.1 0-2 .9-2 2v11h2c0 1.66 1.34 3 3 3s3-1.34 3-3h6c0 1.66 1.34 3 3 3s3-1.34 3-3h2v-5zM6 18.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5m13.5-9 1.96 2.5H17V9.5zm-1.5 9c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5"
            />
          </svg>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1ypmsg6-MuiTypography-root"
          >
            Free Shipping
          </h6>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
        >
          Estimated delivery: 3-5 business days
        </p>
      </div>
    </div>
  </div>
</div>
`;
